_test/test_unique_objects_container/test_unique_references.m

```diff
@@ -0,0 +1,569 @@
+            urc(1) = 'aaaaa';
+            urc(2) = 'bbbb';
+            urc(3) = 'bbbb';
+            function thrower()
+                % unique_objects should be set with a
+                % unique_objects_container, here it is set to a string
+                % which should throw
+                urc.unique_objects = 'bbbb';
+            end
+            assertExceptionThrown(@()thrower, ...
+                'HERBERT:unique_references_container:invalid_argument');
+            
+            uoc = unique_objects_container('baseclass','char');     
+            uoc{1} = 'xxxx';
+            uoc{2} = 'yyyy';
+            uoc{3} = 'zzzz';
+            urc.unique_objects = uoc;
+            
+            uoc = unique_objects_container('baseclass','double');     
+            uoc{1} = 0.111;
+            uoc{2} = 0.222;
+            uoc{3} = 0.333;
+            function wrong_baseclass_thrower()
+                urc.unique_objects = uoc;
+            end
+            assertExceptionThrown(@() wrong_baseclass_thrower, ...
+                'HERBERT:unique_references_container:invalid_argument');
+        end
+        
+        function test_replace_unique_objects_with_cellarray_throw(~)
+            ws = warning('off','HERBERT:unique_references_container:debug_only_argument');
+            clOb = onCleanup(@()warning(ws));
+
+            unique_references_container.global_container('CLEAR','CStrings');
+            urc = unique_references_container('CStrings','char');
+            urc(1) = 'aaaaa';
+            urc(2) = 'bbbb';
+            urc(3) = 'bbbb';
+            function thrower()
+                urc.unique_objects = {'AA','AA'};
+            end
+            assertExceptionThrown(@()thrower, ...
+                'HERBERT:unique_references_container:invalid_argument');
+
+        end
+
+        function test_replace_unique_same_number_works(~)
+            skipTest(['This does not work for unique_reference_containers ', ...
+                      ' as they do not separately store the unique objects.', ...
+                      ' Test kept and skipped as a reminder of this.']);
+            unique_references_container.global_container('CLEAR','CStrings');
+            urc = unique_references_container('CStrings','char');
+            urc(1) = 'aaaaa';
+            urc(2) = 'bbbb';
+            urc(3) = 'bbbb';
+            
+            uoc2 = unique_objects_container('baseclass','char');
+            uoc2(1) = 'dd';
+            uoc2(2) = 'cc';
+            urc.unique_objects = uoc2;
+
+            assertEqual(urc(1),'dd') 
+            assertEqual(urc(2),'cc')
+            assertEqual(urc(3),'cc')
+        end
+
+
+        function test_baseclass_issues(obj)
+            ws = warning('off','HERBERT:unique_references_container:debug_only_argument');
+            clOb = onCleanup(@()warning(ws));
+            ws = warning('off','HERBERT:unique_objects_container:invalid_argument');
+            clOb1 = onCleanup(@()warning(ws));
```

Comment:
Not clear why, the warnings have 2 different IDs

Reply:
The first clob will already reset the warning state to the state it was before any warnings were disabled. The second one is unnecessary. It will also depend on the order the `clOb`s are cleared on which final warning state we get, and I'm not sure MATLAB guarantees that.

Reply:
Ignore this, I apparently misunderstood how warning works. The docs were not clear.